ee/localserver/server.go

```diff
@@ -461,6 +457,12 @@ func (ls *localServer) presenceDetectionHandler(next http.Handler) http.Handler
 			)
 		}
 
+		if !ls.presenceDetectionMutex.TryLock() {
+			http.Error(w, "presence detection already in progress", http.StatusTooManyRequests)
+			return
+		}
+		defer ls.presenceDetectionMutex.Unlock()
```

Comment:
Should we not defer this and just unlock right after call to DetectPresence? I don't think it will unlock until request complete since the func is calling next instead of returning. But maybe it's better to let the whole request complete.

Reply:
I think the defer is fine!